helix-core/src/main/java/org/apache/helix/task/TaskRunner.java

```diff
@@ -203,23 +206,29 @@ private void requestStateTransition(TaskPartitionState state) {
    * @param state     the requested state
    * @return true if the request was persisted, false otherwise
    */
-  private static boolean setRequestedState(HelixDataAccessor accessor, String instance,
+  private boolean setZKCurrentState(HelixDataAccessor accessor, String instance,
       String sessionId, String resource, String partition, TaskPartitionState state) {
     LOG.debug(
-        String.format("Requesting a state transition to %s for partition %s.", state, partition));
+        String.format("Updating current state %s for partition %s.", state, partition));
     try {
       PropertyKey.Builder keyBuilder = accessor.keyBuilder();
       PropertyKey key =
           Boolean.getBoolean(SystemPropertyKeys.TASK_CURRENT_STATE_PATH_DISABLED) ? keyBuilder
               .currentState(instance, sessionId, resource)
               : keyBuilder.taskCurrentState(instance, sessionId, resource);
-      CurrentState currStateDelta = new CurrentState(resource);
-      currStateDelta.setRequestedState(partition, state.name());
 
-      return accessor.updateProperty(key, currStateDelta);
+      String prevState = _stateModel.getCurrentState();
+      CurrentState currentStateDelta = new CurrentState(resource);
+      currentStateDelta.setSessionId(sessionId);
+      currentStateDelta.setStateModelDefRef("Task");
```

Comment:
NIT: we have this "Task" constant. Let's use that instead of hard code it here.

Reply:
THX. Will update.